src/libs/actions/IOU.js

```diff
@@ -386,12 +384,9 @@ function requestMoney(report, amount, currency, payeeEmail, payeeAccountID, part
         },
     };
 
-    let isNewReportPreviewAction = false;
-    let reportPreviewAction = isNewIOUReport ? null : ReportActionsUtils.getReportPreviewAction(chatReport.reportID, iouReport.reportID);
-    if (!reportPreviewAction) {
-        isNewReportPreviewAction = true;
-        reportPreviewAction = ReportUtils.buildOptimisticReportPreview(chatReport.reportID, iouReport.reportID);
-    }
+    let reportPreviewAction = ReportActionsUtils.getReportPreviewAction(chatReport.reportID, iouReport.reportID);
+    const isNewReportPreviewAction = Boolean(!reportPreviewAction);
+    reportPreviewAction = ReportUtils.buildOptimisticReportPreview(chatReport, iouReport, reportPreviewAction);
```

Comment:
This looks odd and will likely introduce a bug when you try to rollback data, since you no longer have the original created date anymore. As in my first comment, I think we should build an optimistic action only if we don't have an existing one.